### ISSUE TITLE
Can't set maxWidth / maxHeight to image

### ISSUE THREAD

**Issue by pseisam** [bpampuch/pdfmake#2462]
It seems like you can't set a maxWidth and maxHeight to an image. So small and big images can't be scaled correctly:

If I have really big images, I can use "fit" to scale them to the page size. But if I do this all images will be scaled. So if I have only a 20x20 pixel image, it will also be scaled to page size and then look terrible.

Thus a feature is needed, which leaves the size of an image if it already fits the page and only gets scaled if it reaches max values for height and width.

**Comment by neissljanik** [comment]
This feature has been implemented in #938, but it doesn't seem to have found its way into the documentation yet. It works like you'd expect, you can specify a `maxHeight` and a `maxWidth` and if the image exceeds any of those dimensions it is scaled down accordingly. So essentially it behaves like `fit` for images that are "too large" while leaving "small enough" images alone.

**Comment by capc0** [comment]
FYI example usage to paste into the playground (http://pdfmake.org/playground.html)

```ts
var dd = {
    content: [
        'maxWidth',
        {
            image: 'sampleImage.jpg',
            // A4 width - page margin left - page margin right
            maxWidth: 595 - 40 - 40
        },
        'maxHeight',
        {
            image: 'sampleImage.jpg',
            maxHeight: 100
        },
        'maxWidth & maxHeight',
        {
            image: 'sampleImage.jpg',
            maxWidth: 100,
            maxHeight: 100,
        },
    ]
}
```

![image](https://github.com/user-attachments/assets/9851e1eb-fcd6-42dc-967c-e87361bd5dfb)
